src/main/java/com/spotify/futures/FuturesExtra.java

```diff
@@ -323,6 +474,23 @@ public Z apply(List<Object> results) {
     Z apply(A a, B b, C c, D d, E e, F f);
   }
 
+  /**
+   * Asynchronously transform the result of 3 {@link ListenableFuture}s a, b, c, d, e and f once
+   * both has completed using {@code function} and complete the returned ListenableFuture at the
+   * point.
+   * Execution is asynchronous in the sense that function returns a Future that may be executed
+   * asynchronously and once it completes it will complete the future that this method returns.
+   *
+   * @param a the first ListenableFuture
+   * @param b the second ListenableFuture
+   * @param c the third ListenableFuture
+   * @param d the fourth ListenableFuture
+   * @param e the fifth ListenableFuture
+   * @param f the sixth ListenableFuture
+   * @param function an AsyncFunction6 implementation, possibly expressed as a java 8 lambda.
+   * @return a ListenableFuture of the result of {@code function}, that completes as soon as
+   * the future that gets returned by the supplied Function6.apply() has completed.
```

Comment:
Oh, now I see. Should have thought of that. I toyed around with a wording that actually referenced the guava functionality that this group of methods wrap, but it became hard to read (you would typically need to be acquainted with the idiom of putting them in an allAsList future to be able to explain it). I think that well written javadoc should help people reading the code in the future to abstract away somewhat complex things.

Unfortunately the wording of upstream doesn't really apply. What do you think about the following wording?
A future that holds result of the function (if all inputs succeeded) or the first failing input's failure (if not)